Userland/Libraries/LibCrypto/BigInt/Algorithms/BitwiseOperations.cpp

```diff
@@ -207,6 +208,15 @@ FLATTEN void UnsignedBigIntegerAlgorithms::shift_left_without_allocation(
     }
 }
 
+FLATTEN void UnsignedBigIntegerAlgorithms::shift_right_without_allocation(
```

Comment:
We just reverted a commit today related to FLATTEN on gcc-13. Have you tested with that compiler to make sure there's not a compile-time regression?

Reply:
We're pretty inconsistent about FLATTEN, ALWAYS_INLINE or nothing with this family of functions.

Reply:
I mainly just followed what the functions around this one do,
I have the feeling that it actually doesn't do anything for most big int functions, as they are distributed over several CUs

I plan on going over more functions to get them to use the AK implementations, and see what happens with the code

Reply:
> I plan on going over more functions to get them to use the AK implementations, and see what happens with the code

Tangentially related: multiplication should give a good speed-up too since it also does a single bit at a time. I also have an fft implementation [stashed](https://github.com/DanShaders/serenity/tree/fft-multiplication) for adequate performance if anyone confuses LibCrypto and gmp.